packages/frontend-2/lib/user/composables/activeWorkspace.ts

```diff
@@ -0,0 +1,50 @@
+import { userActiveWorkspaceQuery } from '~/lib/user/graphql/queries'
+import { useQuery, useMutation } from '@vue/apollo-composable'
+import { setActiveWorkspaceMutation } from '~/lib/user/graphql/mutations'
+import { modifyObjectField, getCacheId } from '~/lib/common/helpers/graphql'
+import type { MaybeNullOrUndefined } from '@speckle/shared'
+
+export const useActiveWorkspace = () => {
+  const { result } = useQuery(userActiveWorkspaceQuery)
+
+  const activeWorkspaceSlug = computed(
+    () => result.value?.activeUser?.activeWorkspace?.slug
+  )
+
+  return {
+    activeWorkspaceSlug
+  }
+}
+
+export const useSetActiveWorkspace = () => {
+  const { activeUser } = useActiveUser()
+  const { mutate } = useMutation(setActiveWorkspaceMutation, {
+    update: (cache, { data }) => {
+      if (!activeUser.value) return
+
+      const newWorkspace = data?.activeUserMutations?.setActiveWorkspace
+
+      modifyObjectField(
+        cache,
+        getCacheId('User', activeUser.value.id),
+        'activeWorkspace',
+        ({ helpers: { ref } }) => {
+          // Handle case where workspace is set to null (personal projects)
+          return newWorkspace ? ref('LimitedWorkspace', newWorkspace.id) : null
+        }
+      )
+    }
+  })
+
+  const setActiveWorkspace = async (options: {
```

Comment:
to ensure either id or slug are passed in, you can type this as `options: {id: string } | { slug: string }`